client/v3/ordering/util.go

```diff
@@ -26,26 +25,18 @@ type OrderViolationFunc func(op clientv3.Op, resp clientv3.OpResponse, prevRev i
 
 var ErrNoGreaterRev = errors.New("etcdclient: no cluster members have a revision higher than the previously received revision")
 
-func NewOrderViolationSwitchEndpointClosure(c clientv3.Client) OrderViolationFunc {
-	var mu sync.Mutex
-	violationCount := 0
+func NewOrderViolationSwitchEndpointClosure(c *clientv3.Client) OrderViolationFunc {
+	violationCount := int32(0)
 	return func(op clientv3.Op, resp clientv3.OpResponse, prevRev int64) error {
-		if violationCount > len(c.Endpoints()) {
+		// Each request is assigned by round-robin load-balancer's picker to a different
```

Comment:
```
// each request is assigned to different endpoints in round-robin manner
// so if we try >5 times, it's very likely that no cluster member has a revision
// higher than the previously specified revision
```

?

This closure seems broken... we're not even using all arguments.

Reply:
I didn't want to change the signature defined by OrderViolationFunc. 

I doubt anyone is using own implementation of this function, but decided to not touch it.

Reply:
Agree.